sdks/python/src/opik/api_objects/dataset/dataset.py

```diff
@@ -53,6 +54,23 @@ def insert(
             for item in items
         ]
 
+        # Remove duplicates if they already exist
+        deduplicated_items = []
+        for item in items:
+            item_hash = utils.compute_content_hash(item)
+
+            if item_hash in self._hash_to_id:
+                if item.id is None or self._hash_to_id[item_hash] == item.id:  # type: ignore
```

Comment:
There's a bug here as basically two items with the same payload and different ID aren't deduped. Not a big deal, but it should be fixed in a follow-up PR.